### PR TITLE
QuickEdit: Add Parent field

### PR DESCRIPTION
**Pull request by gigitux**
## What?

This PR adds the `Parent` field. 
Related issue #64519.

## Testing Instructions

Ensure that `Quick Edit in DataViews` and `Quick Edit in DataViews` are enabled.

1. Visit the Site Editor.
2. Click on pages.
3. Toggle to the table view.
4. Click on the settings.
5. Click on details panel icon.
6. Ensure that the parent field shows the right parent if the page has a parent. Otherwise, it should be rendered: `None`
7. Click on it.
8. Update the parent.
9. Save.
10. Ensure that the parent is set correctly.
11. Perform a smoke test on this field and compare it against the page’s document inspector.

## Screenshots or screencast 
https://github.com/user-attachments/assets/b3f75e5b-61f3-496f-bc2f-eb66891f2e85

### CODE REVIEW

**Comment by oandregal** [comment]
> Warning: Type of PR label mismatch

Thanks for adding the labels! The current setup requires at least one label to be of the `[Type]` kind. I've updated it to have one.

Also, for the PR title: we use this to auto-generate the changelog that is published as part of the release. I understand the package name (`@wordpress/dataviews`) may induce to confusion, but I'd favor a separation between `DataViews` work (for the component that renders list of data) vs `DataForm` or `QuickEdit` for the component that renders forms.

**Comment by gigitux** [comment]
> Thanks for adding the labels! The current setup requires at least one label to be of the `[Type]` kind. I've updated it to have one.

Thanks!

> I understand the package name (@wordpress/dataviews) may induce to confusion, but I'd favor a separation between DataViews work (for the component that renders list of data) vs DataForm or QuickEdit for the component that renders forms.

Sounds great! Thanks for fixing both 🙇

File: packages/fields/src/fields/parent/index.ts
Comment (oandregal) [inline packages/fields/src/fields/parent/index.ts:24]: Thanks for documenting it (README) :)

File: packages/fields/src/fields/parent/index.ts
Comment (oandregal) [inline packages/fields/src/fields/parent/index.ts]: Why disabling sorting? The [endpoint](https://developer.wordpress.org/rest-api/reference/pages/#list-pages) we take the data from supports sorting by `parent`.

File: packages/fields/src/fields/parent/parent-view.tsx
Comment (oandregal) [inline packages/fields/src/fields/parent/parent-view.tsx:32]: This works for me: we could land as it is and then gather feedback from folks. Though I wonder if there's a better default for this because `None` could be an actual title, so it may be confusing. Perhaps `—` like wp-admin uses in some places?

File: packages/fields/src/fields/parent/utils.ts
Comment (oandregal) [inline packages/fields/src/fields/parent/utils.ts:12]: There's a `getItemTitle` utilility in [actions](https://github.com/WordPress/gutenberg/blob/ce7f1d857b5a35b430c8a37b604d57e2402276da/packages/fields/src/actions/utils.ts#L33). Can we reuse that one?

Comment (oandregal) [reply packages/fields/src/fields/parent/utils.ts:12]: Interestingly, it appears that every usage of `decodeEntities` in the fields package can be replaced by just calling that utility?

In [some situations](https://github.com/WordPress/gutenberg/blob/ce7f1d857b5a35b430c8a37b604d57e2402276da/packages/fields/src/actions/delete-post.tsx#L98), we are running it twice:

```js
decodeEntities( decodeEntities( '...' ) );
```

File: packages/fields/src/fields/parent/parent-edit.tsx
Comment (oandregal) [inline packages/fields/src/fields/parent/parent-edit.tsx:51]: Interesting how using TypeScript here forced us to declare the Term API (parent obligatory), making the prop redundant here compared to the [original](https://github.com/WordPress/gutenberg/blob/ce7f1d857b5a35b430c8a37b604d57e2402276da/packages/editor/src/utils/terms.js#L17) that used:

```js
{
  children: [],
  parent: null,
  ...term,
}
```

File: packages/fields/src/fields/parent/parent-edit.tsx
Comment (oandregal) [inline packages/fields/src/fields/parent/parent-edit.tsx]: It doesn't seem `parentOptions` is used anywhere else, so can we make this part of the `useMemo` above? Otherwise, this is creating a new set of options in every render.

File: packages/fields/src/fields/parent/parent-edit.tsx
Comment (oandregal) [inline packages/fields/src/fields/parent/parent-edit.tsx:119]: It's a bit weird to bind `data.parent` to two different variables here, but I understand why you did it: to make the `useSelect` below as similar as possible to the [original](https://github.com/WordPress/gutenberg/blob/ce7f1d857b5a35b430c8a37b604d57e2402276da/packages/editor/src/components/page-attributes/parent.js#L64). I think it works nicely for this use case, though I wouldn't mind consolidating into a single one either — whatever you feel is right. Though, if you want to keep it like this, it'd be good to add a comment to explain the why so other readers don't have to discover it themselves.

**Review by oandregal — APPROVED** [review]
I've left some comments, but they're a bit minor so I wanted to pre-approve for you to merge when ready.

This works well and it is a well-scoped PR, thanks!

File: packages/fields/src/fields/parent/parent-view.tsx
Comment (gigitux) [reply packages/fields/src/fields/parent/parent-view.tsx:32]: I would proceed in this way. 

`None` is used in the Document Inspector panel:

image

Comment (oandregal) [reply packages/fields/src/fields/parent/parent-view.tsx:32]: Oh, nice, it makes more sense then.

File: packages/fields/src/fields/parent/utils.ts
Comment (gigitux) [reply packages/fields/src/fields/parent/utils.ts:12]: What do you think about?

```
export function getTitleWithFallbackName( post: BasePost ) {
    const title = getTitle(post)
    return title.length > 0 ? title : `#${ post?.id } (${ __( 'no title' ) })`;
```

File: packages/fields/src/fields/parent/parent-edit.tsx
Comment (gigitux) [reply packages/fields/src/fields/parent/parent-edit.tsx]: Fixed with 4526f965cd9d2fbc8e2539a5fb11e0f188c9df1f

File: packages/fields/src/fields/parent/parent-edit.tsx
Comment (gigitux) [reply packages/fields/src/fields/parent/parent-edit.tsx:119]: >  though I wouldn't mind consolidating into a single one either

I agree with you! Improved with af5dec22041925566f104fd692d99feb4099141c

File: packages/fields/src/fields/parent/index.ts
Comment (gigitux) [reply packages/fields/src/fields/parent/index.ts]: Fixed with c7f30b90584c96ae728581f14bdcbdd1aa75865c

File: packages/fields/src/fields/parent/utils.ts
Comment (oandregal) [reply packages/fields/src/fields/parent/utils.ts:12]: I guess my point was that we didn't need `getTitleWithFallbackName` and we could reuse `getItemTitle`, unless I'm missing something?

Comment (gigitux) [reply packages/fields/src/fields/parent/utils.ts:12]: This function is simply a way to centralize the fallback return value. Otherwise, we need to implement it multiple times.

```
const title = getTitle(post)
title.length > 0 ? title : `#${ post?.id } (${ __( 'no title' ) })`;
```

In the specific, we should do repeat this check three times:

image

Comment (oandregal) [reply packages/fields/src/fields/parent/utils.ts:12]: I guess what I don't understand is why we have different ways to get the title from a Post across the app. Anyway, don't feel this PR should be blocked by this comment. This can be looked at separately.

**Pull request merged**
